File: frontend/src/types/repository.ts
/**
 * Repository Types
 */

export interface Repository {
  id: string;
  github_id: number;
  owner_username: string;
  name: string;
  full_name: string;
  description?: string;
  is_private: boolean;
  default_branch: string;
  url?: string; // GitHub URL
  html_url?: string;
  clone_url?: string;
  language?: string;
  stargazers_count?: number;
  forks_count?: number;
  open_issues_count?: number;
  github_created_at?: string;
  github_pushed_at?: string;
  created_at: string;
  updated_at?: string;
}

export interface RepositoryCreate {
  name: string;
  description?: string;
  private?: boolean;
}

export interface RepositoryUpdate {
  description?: string;
  default_branch?: string;
}

File: backend/src/models/repository/repository.py
"""
Repository model - GitHub repositories
"""

from pydantic import BaseModel, Field

from ..base import BaseEntity, BaseSemanticModel


class Repository(BaseEntity, BaseSemanticModel):
    """GitHub repository model"""

    full_name: str = Field(..., description="owner/repo")
    owner_username: str = Field(..., description="Owner username")
    github_id: int | None = Field(None, description="GitHub repository ID")
    default_branch: str = Field(default="main", description="Default branch")
    is_private: bool = Field(default=False, description="Is private repository")
    github_created_at: str | None = Field(None, description="Creation timestamp from GitHub")
    github_pushed_at: str | None = Field(None, description="Last push timestamp from GitHub")

    # Stats
    open_issues_count: int = Field(default=0, description="Number of open issues")


class RepositoryCreate(BaseModel):
    """Data needed to create a repository"""

    name: str
    description: str | None = None
    private: bool = False


class RepositoryUpdate(BaseModel):
    """Data for updating a repository"""

    name: str | None = None
    description: str | None = None
    default_branch: str | None = None
    open_issues_count: int | None = None
